### PR TITLE
Match oculus system browser controller mapping

### PR DESCRIPTION
**Pull request by bluemarvin**

### CODE REVIEW

File: app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp [hunk 1]
```diff
@@ -922,19 +922,27 @@ struct DeviceDelegateOculusVR::State {
...
 
-          controller->SetButtonState(i, ControllerDelegate::BUTTON_APP, -1, menuPressed, menuPressed);
+
           controller->SetButtonState(i, ControllerDelegate::BUTTON_X, 3, xPressed, xTouched);
           controller->SetButtonState(i, ControllerDelegate::BUTTON_Y, 4, yPressed, yTouched);
+
+          if (renderMode != device::RenderMode::Immersive) {
+            controller->SetButtonState(i, ControllerDelegate::BUTTON_APP, -1, yPressed, yTouched);
+          } else {
+            controller->SetButtonState(i, ControllerDelegate::BUTTON_APP, -1, menuPressed, menuPressed);
```
Comment (cvan) [inline app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: could we have the Menu button also recognised as Exit in Fullscreen Movie Mode?

Comment (cvan) [reply app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: FWIW, Oculus Browser uses Y and B to Exit Fullscreen and its Movie Mode?

File: app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp [hunk 2]
```diff
@@ -922,19 +922,27 @@ struct DeviceDelegateOculusVR::State {
...
           const bool bPressed = (state[i].controllerState.Buttons & ovrButton_B) != 0;
           const bool bTouched = (state[i].controllerState.Touches & ovrTouch_B) != 0;
-          const bool menuPressed = (state[i].controllerState.Buttons & ovrButton_Enter) != 0;
 
-          controller->SetButtonState(i, ControllerDelegate::BUTTON_APP, -1, menuPressed, menuPressed);
           controller->SetButtonState(i, ControllerDelegate::BUTTON_A, 3, aPressed, aTouched);
-          controller->SetButtonState(i, ControllerDelegate::BUTTON_B, 4, bPressed, bTouched);
+          controller->SetButtonState(i, ControllerDelegate::BUTTON_Y, 4, bPressed, bTouched);
+
+          if (renderMode != device::RenderMode::Immersive) {
```
Comment (cvan) [inline app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: in Oculus Browser, `X` & `A` buttons toggle the Media Controls. could we do the same for ours?

File: app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp (hunk 1 above)
Comment (cvan) [reply app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: tested both again. this behaviour is same as Oculus Browser's, so you can ignore this comment. but see comment below about toggling the Media Controls.

File: app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp (hunk 2 above)
Comment (bluemarvin) [reply app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: Let me see how easy it is.

File: app/src/main/cpp/BrowserWorld.cpp [hunk 3]
```diff
@@ -453,11 +453,19 @@ BrowserWorld::State::UpdateControllers(bool& aRelayoutWidgets) {
...
+      VRBrowser::HandleMotionEvent(0, controller.index, (jboolean) pressed, 0.0f, 0.0f);
       controller.widget = 0;
 
     } else if (wasPressed != pressed) {
-      VRBrowser::HandleMotionEvent(0, controller.index, pressed, 0.0f, 0.0f);
+      VRBrowser::HandleMotionEvent(0, controller.index, (jboolean) pressed, 0.0f, 0.0f);
+    } else if (vrVideo != nullptr) {
+      const bool togglePressed = controller.buttonState & ControllerDelegate::BUTTON_X ||
+                                 controller.buttonState & ControllerDelegate::BUTTON_A;
+      const bool toggleNotPressed = controller.lastButtonState & ControllerDelegate::BUTTON_X ||
```
Comment (MortimerGoro) [inline app/src/main/cpp/BrowserWorld.cpp]: nit: `toggleWasPressed` may be a better name

Comment (bluemarvin) [reply app/src/main/cpp/BrowserWorld.cpp]: done

**Pull request merged**